src/libraries/System.IO.FileSystem/tests/LegacyTests/LegacySwitchTests.cs

```diff
@@ -20,7 +20,7 @@ public static void LegacySwitchIsHonored()
                     .GetField("_strategy", BindingFlags.NonPublic | BindingFlags.Instance)
                     .GetValue(fileStream);
 
-                Assert.DoesNotContain(strategy.GetType().FullName, "Legacy");
+                Assert.Contains(strategy.GetType().FullName, "Legacy");
```

Comment:
Can we also change the "Legacy" naming?

Reply:
> Can we also change the "Legacy" naming?

Of the internal types and properties? Sure, do you have any good suggestions? `Net5CompatFileStreamStrategy`? `BackwardCompatibileFileStreamStrategy`?

Reply:
Yes,  they show up in exception stack traces, for example. Your first suggestion sounds good.

Reply:
> Your first suggestion sounds good.

Ok, I've added the rename to the TODO list of this PR. 

@carlossanlop @Jozkee please let me know if you have a better idea for a new name

Reply:
I've sent #50470 with the rename